packages/nextjs/utils/scaffold-eth/getMetadata.ts

```diff
@@ -30,5 +30,12 @@ export const getMetadata = ({
       description: description,
       images: [imageUrl],
     },
+    other: {
+      "fc:frame": "vNext",
+      "fc:frame:image": imageUrl,
+      "fc:frame:button:1": "Open my 🏗️ Scaffold-ETH 2 dApp",
+      "fc:frame:button:1:action": "link",
+      "fc:frame:button:1:target": baseUrl,
```

Comment:
NITPICK: Should we accept this args in function and maybe have default value ? 

So that people can have respective button / URL for their route 

EG: I have `/builders` page and I am using `getMetadata` function in it

I can maybe have `getMetadata({framRelativeRoute: `/builders`, frameBtnLabel: "Open Builders page"})` ? 

Usefull when I share `https://frames-se2.vercel.app/builders`  lol but not sure if it makes that much sense

Reply:
I think it makes sense, and having a default **frameBtnLabel**.

Reply:
> EG: I have /builders page and I am using getMetadata function in it
> I can maybe have getMetadata({framRelativeRoute: "/builders", frameBtnLabel: "Open Builders page"}) ?
> Usefull when I share https://frames-se2.vercel.app/builders lol but not sure if it makes that much sense

Do you think this is a use case for Frames? From what I've seen, people create a _single_ frame for an app (typically more complex than what we are doing here). Abstracting "framRelativeRoute" + "frameBtnLabel"  seems a bit too much.

If they want to build a real frame, they'll need to create a backend route with the dynamic steps, so these abstractions won't serve much.

My initial idea was to make a simple/silly "frame" so people could see it out of the box, and tweak as needed. But merging this PR might not be a good idea after all :D

Reply:
Yes, you have a point.

We could:

a) leave it as it's 
b) as it's, but only add it for the homepage as a example
c) make it configurable as you were suggesting

I always try to push on things like (c) to keep it minimal (especially in things that doesn't seem too important). And yes, it's a tiny thing to add... but tiny things multiply over time and makes DX worse.

In any case, open to to add it!

My vote will be for (b), and then (a)

Thanks Shiv.

Reply:
Yes, a or b is good enough for now.

Reply:
also voting for b (for me b > c > a)

Reply:
There is a problem with **b**

Currently, the homepage has "use client" (because of the account hook), so we can't export the metadata from there (only available for server-side pre-rendered pages).

I don't think is worth it to add a sub-component in the homepage (to move the `use client` there) just for the sake of this PR.

Alternatives:
- Leave the `other` in the getMetadata function
- Add the `other` param in the getMetadata function, and add the frame in `layout.tsx` (passing all the frame data in the other param)


What do you guys think?

Reply:
I think we can leave getMetadata as is, it is not a big deal.

Reply:
> Umm, I would vote to not add this frame thing at all 😅. Keeps the code simple and predictable.

Yeah, I understand why you feel this. Inconsistent behavior in the different routes.

So opted to remove it from this PR for now. We can discuss later on.